src/main/java/services/gorest/stepdefinition/UserStepsDefinition.java

```diff
@@ -0,0 +1,101 @@
+})
+public class UserStepsDefinition {
+
+    @Steps
+    private CreateUser createUser;
+
+    @Steps
+    private GetUser getUser;
+
+    @Steps
+    private DeleteUser deleteUser;
+
+    @Steps
+    private UpdateUser updateUser;
+
+    @Steps
+    private CommonValidations commonValidations;
+
+    @Steps
+    private ReusableMethods reusableMethods;
+
+
+    @When("^I create a new user with email (.*), first name (.*), last name (.*) and gender (.*)$")
+    public void whenCreateUser(String email, String firstName, String lastName, String gender) {
+        Response response = createUser.whenCreateNewUser(email, firstName, lastName, gender);
+        setSessionVariable(VAR_USER, response.as(GetUserResponse.class).getResult());
+        setSessionVariable(VAR_USER_ID, response.as(GetUserResponse.class).getResult().getId());
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+    @Then("^I check that the status code from the response body is (.*)$")
+    public void thenCheckStatusCodeFromResponseBody(int statusCode) {
+        commonValidations.validateResponseStatusCode(getSessionVariable(VAR_RESPONSE), statusCode);
+    }
+
+    @When("^I retrieve a single user with the id: (.*)$")
+    public void whenGetUserFromResponse(String userId) {
+        reusableMethods.setSessionVariableAsExpected(userId, VAR_USER_ID);
+        Response response = getUser.getUserById(getSessionVariable(VAR_USER_ID));
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+    @When("^I delete a single user based on id: (.*)$")
+    public void whenDeleteUserById(String userId) {
+        reusableMethods.setSessionVariableAsExpected(userId, VAR_USER_ID);
+        Response response = deleteUser.deleteUserById(getSessionVariable(VAR_USER_ID));
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+    @When("^I update the user with id (.*) with the new last name (.*)$")
+    public void whenUpdateUserById(String userId, String lastName) {
+        reusableMethods.setSessionVariableAsExpected(userId, VAR_USER_ID);
+        Response response = updateUser.whenUpdateUsersLastNameUsingId(getSessionVariable(VAR_USER_ID), lastName);
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+    @When("^I update the user with id (.*) with first name (.*), last name (.*), email (.*), status (.*) and gender (.*)$")
+    public void whenUpdateAllUsersDetailsById(String userId, String firstName, String lastName, String email, String status, String gender) {
+        reusableMethods.setSessionVariableAsExpected(userId, VAR_USER_ID);
+        Response response = updateUser.whenUpdateAllUserDetailsById(getSessionVariable(VAR_USER_ID), firstName, lastName, email, status, gender);
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+    @After("@UserSmoke")
+    public void tearDownDeleteUser(Scenario scenario) {
+        if (!scenario.getName().equals("Deleting a users details")) {
+            if (getSessionVariable(VAR_USER_ID) != null) {
```

Comment:
you cannot rely on the fact that even if you set the session variable with the id, the resource wasn't deleted before in other test. best approach here is to get the resource first and check it exists and after that delete it.